Jellyfin.Server/Migrations/MigrationRunner.cs

```diff
@@ -111,38 +119,60 @@ private static void HandleStartupWizardCondition(IEnumerable<IMigrationRoutine>
 
-            for (var i = 0; i < migrations.Length; i++)
+            string? migrationKey = null;
+            if (migrationsToBeApplied.Any(f => f is IDatabaseMigrationRoutine) && jellyfinDatabaseProvider is not null)
             {
-                var migrationRoutine = migrations[i];
-                if (appliedMigrationIds.Contains(migrationRoutine.Id))
-                {
-                    logger.LogDebug("Skipping migration '{Name}' since it is already applied", migrationRoutine.Name);
-                    continue;
-                }
-
-                logger.LogInformation("Applying migration '{Name}'", migrationRoutine.Name);
-
+                logger.LogInformation("Database backup is performed");
                 try
                 {
-                    migrationRoutine.Perform();
+                    migrationKey = await jellyfinDatabaseProvider.MigrationBackupFast(CancellationToken.None).ConfigureAwait(false);
                 }
-                catch (Exception ex)
+                catch (NotImplementedException)
                 {
-                    logger.LogError(ex, "Could not apply migration '{Name}'", migrationRoutine.Name);
-                    throw;
+                    logger.LogWarning("Could not perform backup of database before migration because provider does not support it");
                 }
+            }
+
+            try
+            {
+                foreach (var migrationRoutine in migrationsToBeApplied)
+                {
+                    logger.LogInformation("Applying migration '{Name}'", migrationRoutine.Name);
+
+                    try
+                    {
+                        migrationRoutine.Perform();
+                    }
+                    catch (Exception ex)
+                    {
+                        logger.LogError(ex, "Could not apply migration '{Name}'", migrationRoutine.Name);
+                        throw;
+                    }
 
-                // Mark the migration as completed
-                logger.LogInformation("Migration '{Name}' applied successfully", migrationRoutine.Name);
-                migrationOptions.Applied.Add((migrationRoutine.Id, migrationRoutine.Name));
-                saveConfiguration(migrationOptions);
-                logger.LogDebug("Migration '{Name}' marked as applied in configuration.", migrationRoutine.Name);
+                    // Mark the migration as completed
+                    logger.LogInformation("Migration '{Name}' applied successfully", migrationRoutine.Name);
+                    migrationOptions.Applied.Add((migrationRoutine.Id, migrationRoutine.Name));
+                    saveConfiguration(migrationOptions);
+                    logger.LogDebug("Migration '{Name}' marked as applied in configuration.", migrationRoutine.Name);
+                }
+            }
+            catch (System.Exception) when (migrationKey is not null && jellyfinDatabaseProvider is not null)
```

Comment:
It's impossible for jellyfinDatabaseProvider to be null, no?
```suggestion
            catch (System.Exception) when (migrationKey is not null)
```